ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,290 @@
+
+Usage
+-----
+
+.. code-block:: python
+
+    from opentelemetry import trace
+    from opentelemetry.ext.cloud_trace import CloudTraceSpanExporter
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import SimpleExportSpanProcessor
+
+    trace.set_tracer_provider(TracerProvider())
+
+    cloud_trace_exporter = CloudTraceSpanExporter()
+    trace.get_tracer_provider().add_span_processor(
+        SimpleExportSpanProcessor(cloud_trace_exporter)
+    )
+    tracer = trace.get_tracer(__name__)
+    with tracer.start_as_current_span("foo"):
+        print("Hello world!")
+
+
+API
+---
+"""
+
+import logging
+from typing import Any, Dict, List, Sequence, Tuple
+
+import google.auth
+from google.cloud.trace_v2 import TraceServiceClient
+from google.cloud.trace_v2.proto.trace_pb2 import AttributeValue
+from google.cloud.trace_v2.proto.trace_pb2 import Span as ProtoSpan
+from google.cloud.trace_v2.proto.trace_pb2 import TruncatableString
+from google.rpc.status_pb2 import Status
+
+import opentelemetry.trace as trace_api
+from opentelemetry.sdk.trace import Event
+from opentelemetry.sdk.trace.export import Span, SpanExporter, SpanExportResult
+from opentelemetry.util import types
+
+logger = logging.getLogger(__name__)
+
+# Max length is 128 bytes for a truncatable string.
+MAX_LENGTH = 128
+
+
+class CloudTraceSpanExporter(SpanExporter):
+    """Cloud Trace span exporter for OpenTelemetry.
+
+    Args:
+        project_id: ID of the cloud project that will receive the traces.
+        client: Cloud Trace client. If not given, will be taken from gcloud
+            default credentials
+    """
+
+    def __init__(
+        self, project_id=None, client=None,
+    ):
+        self.client = client or TraceServiceClient()
+        if not project_id:
+            _, self.project_id = google.auth.default()
+        else:
+            self.project_id = project_id
+
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        """Export the spans to Cloud Trace.
+
+        See: https://cloud.google.com/trace/docs/reference/v2/rest/v2/projects.traces/batchWrite
+
+        Args:
+            spans: Tuple of spans to export
+        """
+        cloud_trace_spans = []
+        for span in self.translate_to_cloud_trace(spans):
+            try:
+                cloud_trace_spans.append(self.client.create_span(**span))
+            # pylint: disable=broad-except
+            except Exception as ex:
+                logger.warning("Error %s when creating span %s", ex, span)
+
+        try:
+            self.client.batch_write_spans(
+                "projects/{}".format(self.project_id), cloud_trace_spans,
+            )
+        # pylint: disable=broad-except
+        except Exception as ex:
+            logger.warning("Error while writing to Cloud Trace: %s", ex)
+            return SpanExportResult.FAILURE
+
+        return SpanExportResult.SUCCESS
+
+    def translate_to_cloud_trace(
```

Comment:
should this be a public method? feels like protected to me, I'd argue most exporters should just provide implementations of the ABC methods.